src/common/Formatter.cc

```diff
@@ -581,12 +581,11 @@ std::string XMLFormatter::escape_xml_str(const char *str)
   return std::string(&escaped[0]);
 }
 
-char XMLFormatter::to_lower_underscore(const char c)
+char XMLFormatter::to_lower_underscore(char c) const
 {
-  if (c == ' ') {
-    if (m_underscored)
+  if (m_underscored && c == ' ') {
       return '_';
-  } else if (m_lowercased) {
+  } else if (m_lowercased && std::isupper(c)) {
```

Comment:
IMO, i don't think this improves the readability. it just hurts the performance here. but yeah, probably the performance here is not a bottleneck...